### DISCUSSION TITLE
Blazor: Integrate CSS (including preprocessors) into .razor files

### DISCUSSION THREAD

**Discussion by rosyatrandom** [dotnet/csharplang#7091] [Language Ideas]
Inspired by JavaScript frameworks with single file components, such as Vue or Svelte,  this proposal is to allow .razor files to contain CSS blocks with preprocessors. (In fact, pretty much all of the CSS functionality in Vue, such as scoping and modules, should be portable)

https://vuejs.org/api/sfc-css-features.html CSS blocks in Vue SFCs
https://vue-loader.vuejs.org/guide/pre-processors.html#sass CSS preprocessors in Vue SFCs

Razor files can already be written in either a split, code-behind, style or a code-with-template style that closely follows the SFCs above; there is no good reason to not to have the choice to include CSS within them.

**Comment by HaloFour** [comment]
This sounds like it belongs on the [ASP.NET](https://github.com/dotnet/aspnetcore) repo?  Razor syntax isn't a part of the C# language.
